src/server/kube-store.js

```diff
@@ -128,7 +138,89 @@ class KubeStore {
                 return;
             }
             default: {
-                throw new Error(`Could not patch noobaa, got ${JSON.stringify(body)}`);
+                throw new Error(`Could not patch noobaa, (status code: ${status_code}) got ${JSON.stringify(body)}`);
+            }
+        }
+    }
+
+    async create_backingstore(new_store) {
+        await this._init();
+        const path = get_backingstores_path(this._k8s_namespace);
+        const { status_code, body } = await this._make_k8s_api_request('POST', path, new_store);
+        switch (status_code) {
+            case 201: {
+                return body;
+            }
+            case 404: {
+                return null;
+            }
+            default: {
+                throw new Error(`Could not create backingstore, (status code: ${status_code}) got ${JSON.stringify(body)}`);
+            }
+        }
+    }
+
+    async delete_backingstore(name) {
+        await this._init();
+        const path = get_backingstores_path(this._k8s_namespace) + `/${name}`;
+        const { status_code, body } = await this._make_k8s_api_request('DELETE', path);
+        switch (status_code) {
+            case 200: {
+                return body;
```

Comment:
I don't think delete operation returns a body. I think in this case you should return a bool,
true for 200 and false for 404 or other codes that we want to take care about. All other codes will default to throwing.